src/java/src/main/java/com/dnanexus/DXEnvironment.java

```diff
@@ -511,6 +597,56 @@ public int getSocketTimeout() {
         return this.socketTimeout;
     }
 
+    /**
+     * Parse the proxy definition. This could take one of several forms:
+     *    localhost:3128
+     *    http://localhost:3128
+     *    https://localhost:3128
+     *    https://dnanexus:welcome@localhost:3128
+     */
+    public ProxyDesc parseProxyDefinition(String proxyDfn, String proxyMethod, String proxyDomain) {
```

Comment:
Does this have to be public?

Reply:
Correct, it doesn't have to be. I'll make it private.